meson.build

```diff
@@ -429,21 +429,24 @@ cndp_a_name = 'libcndp.a'
 cndp_so_name = 'libcndp.so'
 
 mklib = find_program('tools/mklib.sh')
-run_command(mklib, cndp_a_name, libcndp_a, check: true)
-run_command(mklib, cndp_so_name, libcndp_so, check: true)
-
-cndp_a = custom_target('libcndp_a_target',
-    output: cndp_a_name,
-    input: join_paths('/tmp', cndp_a_name),
-    command:[find_program('cp'), '@INPUT@', cndp_a_name],
-    install_dir: join_paths('lib', 'x86_64-linux-gnu'),
-    install: true)
-cndp_so = custom_target('libcndp_so_target',
-    output: cndp_so_name,
-    input: join_paths('/tmp', cndp_so_name),
-    command:[find_program('cp'), '@INPUT@', cndp_so_name],
-    install_dir: join_paths('lib', 'x86_64-linux-gnu'),
-    install: true)
+
+if use_static_libs
+    run_command(mklib, cndp_a_name, libcndp_a, check: true)
+    cndp_a = custom_target('libcndp_a_target',
+        output: cndp_a_name,
+        input: join_paths('/tmp', cndp_a_name),
+        command:[find_program('cp'), '@INPUT@', cndp_a_name],
+        install_dir: join_paths('lib', 'x86_64-linux-gnu'),
+        install: true)
+else
+    run_command(mklib, cndp_so_name, libcndp_so, check: true)
+    cndp_so = custom_target('libcndp_so_target',
+        output: cndp_so_name,
+        input: join_paths('/tmp', cndp_so_name),
+        command:[find_program('cp'), '@INPUT@', cndp_so_name],
+        install_dir: join_paths('lib', 'x86_64-linux-gnu'),
+        install: true)
+endif
```

Comment:
Did we need to create a libcndp_pmds.a file similar to libcndp.so/.a ?

Reply:
Good point. But I think mostly applications doesn't need to use all pmds. So linking all pmds as whole-archive (using libcndp_pmd.a) is unnecessary IMO. So would selecting liking of required PMDs as whole-archive based on application need seems better.

Reply:
We have so few PMDs it would only add about 64K to the image.

Reply:
yes, please create it.

Reply:
Is tun also a pmd that needs to be included in `libcndp_pmds.a` ?

Reply:
No Tun is how we allocate the interfaces, but it is called out for the PMD and it is not a PMD

Reply:
Updated PR